controller/src/main/java/io/pravega/controller/server/rpc/auth/AuthHelper.java

```diff
@@ -67,17 +74,17 @@ public String checkAuthorizationAndCreateToken(String resource, AuthHandler.Perm
     private String createDelegationToken(String resource, AuthHandler.Permissions expectedLevel, String tokenSigningKey) {
         if (isAuthEnabled) {
             Map<String, Object> claims = new HashMap<>();
-
             claims.put(resource, String.valueOf(expectedLevel));
 
-            return Jwts.builder()
-                       .setSubject("segmentstoreresource")
-                       .setAudience("segmentstore")
-                       .setClaims(claims)
-                       .signWith(SignatureAlgorithm.HS512, tokenSigningKey.getBytes())
-                       .compact();
+            return new JsonWebToken("segmentstoreresource",
```

Comment:
A majority of the parameters of `JsonWebToken` are mandatory, so using a builder here would still require the calling code to specify most of them upfront when invoking the builder (as shown below. Also, there is no use case for a step-by-step construction as of now. So, I preferred the (arguably) simpler telescopic constructor pattern instead of the builder, here. 

The following two examples show how the calling code in current use'd look like if we were to use a builder instead:

```java
JsonWebToken.builder("subject", "audience", signingKey, ttl)
     .resourcePermissionClaims(claims)
     .build() 
```
or 
```java
JsonWebToken.builder("subject", "audience", signingKey).build() 
```
What do you say?